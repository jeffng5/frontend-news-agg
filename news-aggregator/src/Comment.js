import React, { useState, useEffect } from 'react'
import "./css/forum.css"
import { Helpers } from './helpers'



//component to add comment 
const Comment = ({id}) => {
  const username = localStorage.getItem('username')
  const [comment, setComment] = useState([])
  const [datetime, setDateTime] = useState(new Date())
  const [addedClause, setAddedClaus] = useState([])

  console.log(username)




  async function addComment(e) {
    e.preventDefault();
    setComment(e.target.value);
    setDateTime(new Date())
    //helpers function to post comment
    const res = await Helpers.postComment(username,  comment, id, datetime)
    console.log(res)
    // note to user that comment has been added
    setAddedClaus('comment has been added')
    
 }
//handleChange function
  const handleChange= (e) => {
    e.preventDefault();
    setComment(e.target.value)
    console.log(comment)
    

 }
return (
    <>

<form>

<textarea placeholder = 'Type comment here...' name='message' rows='5' cols='50' onChange={handleChange}>
    

</textarea>
</form>
<button className= 'add-comment' onClick={addComment}>Add Comment</button><h4>{addedClause}</h4>
</>
)
}
export default Comment