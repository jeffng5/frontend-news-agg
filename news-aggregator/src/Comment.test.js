import React from 'react';
import { render } from '@testing-library/react';
import Comment from './Comment'

it('renders without crashing', function() {
  render(<Comment />);
})

//snapshot test
it("matches snapshot", function() {
    const {asFragment} = render(<Comment />);
    expect(asFragment()).toMatchSnapshot();
  });

test('testing component', ()=> {
    const {getByText} = render( <Comment />);
    getByText('Add Comment')
})