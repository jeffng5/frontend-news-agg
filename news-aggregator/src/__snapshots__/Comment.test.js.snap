// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <textarea
      cols="50"
      name="message"
      placeholder="Type comment here..."
      rows="5"
    />
  </form>
  <button
    class="add-comment"
  >
    Add Comment
  </button>
  <h4 />
</DocumentFragment>
`;
